File: app/src/main/java/ie/dublinbuspal/android/view/favourite/FavouritesPresenter.java
package ie.dublinbuspal.android.view.favourite;

import com.hannesdorfmann.mosby3.mvp.MvpPresenter;

import java.util.List;

import ie.dublinbuspal.model.favourite.FavouriteStop;

public interface FavouritesPresenter extends MvpPresenter<FavouritesView> {

    void onResume();

    void onPause();

    void onPause(List<FavouriteStop> favourites);

}

File: app/src/main/kotlin/ie/dublinbuspal/android/view/favourites/FavouritesPresenter.kt
package ie.dublinbuspal.android.view.favourites

import com.hannesdorfmann.mosby3.mvp.MvpPresenter

interface FavouritesPresenter : MvpPresenter<FavouritesView> {

    fun onResume()

    fun onPause(shouldSaveFavourites: Boolean)

    fun onReorderFavourites(position1: Int, position2: Int)

    fun onFinishedReorderFavourites()

}
